scripts/zones/Promyvion-Vahzl/mobs/Provoker.lua

```diff
@@ -0,0 +1,67 @@
+-----------------------------------
+-- Area: Promyvion - Vahzl
+--   NM: Provoker
+-----------------------------------
+require("scripts/globals/status")
+require("scripts/globals/magic")
+-----------------------------------
+
+function onMobInitialize(mob)
+    mob:addMod(tpz.mod.DOUBLE_ATTACK, 25)
+    mob:addMod(tpz.mod.ACC, 50)
+    mob:setMobMod(tpz.mobMod.IDLE_DESPAWN, 180)
+    mob:setMobMod(tpz.mobMod.ADD_EFFECT, 1)
+end
+
+function onMobFight(mob, target)
+
+    local changeTime = mob:getLocalVar("changeTime")
+    local element = mob:getLocalVar("element")
+
+    if (changeTime == 0) then
+        mob:setLocalVar("changeTime", math.random(2, 3)*15)
+        return
+    end
+    if (mob:getBattleTime() >= changeTime) then
+        local newelement = element
+        while (newelement == element) do
+            newelement = math.random(1, 8)
+        end
+        if (element ~= 0) then
+            mob:delMod(tpz.magic.absorbMod[element], 100)
+        end
+
+        mob:useMobAbility(624)
+        mob:addMod(tpz.magic.absorbMod[newelement], 100)
+        mob:setLocalVar("changeTime", mob:getBattleTime() + math.random(2, 3)*15)
+        mob:setLocalVar("element", newelement)
+    end
+end
+
+function onAdditionalEffect(mob, target, damage)
+
+    local element = mob:getLocalVar("element")
+
+    if (element ~= 0) then
+        if (element == 1) then
+            return tpz.mob.onAddEffect(mob, target, damage, tpz.mob.ae.ENFIRE, {chance = 1000})
```

Comment:
suggestion:
```lua
function onAdditionalEffect(mob, target, damage)
    local element = mob:getLocalVar("element")
    local index =
    {
        [1] = tpz.mob.ae.ENFIRE
        [2] = tpz.mob.ae.ENSTONE
        [3] = tpz.mob.ae.ENWATER
        [4] = tpz.mob.ae.ENAERO
        [5] = tpz.mob.ae.ENBLIZZARD
        [6] = tpz.mob.ae.ENTHUNDER
        [7] = tpz.mob.ae.ENLIGHT
        [8] = tpz.mob.ae.ENDARK
    }
    if index[element] then
        return tpz.mob.onAddEffect(mob, target, damage, index[element], {chance = 1000})
    else
        return 0, 0, 0 -- Just in case its somehow not got a variable set
    end
end
```
then we don't have to do all that if/elseif. The else in my sample above can be ripped out if the mob spawns in with the en-effect already active for its element. I do not know if retail does that I have no looked at the capture. if it does we probably want to set a random element at spawn time just so there is no tiny milliseconds gap between its AI trying to swing at you and the variable being set by onMobFight to throw an explainable nil that seemingly can't  be reproduced when you try to reproduce it later :)